R/adjacency.R

```diff
@@ -377,157 +375,67 @@ mysummary <- function(x) {
-      dd <- el[2:nrow(el), 1] == el[1:(nrow(el) - 1), 1] &
-        el[2:nrow(el), 2] == el[1:(nrow(el) - 1), 2]
-      dd <- which(dd)
-      if (length(dd) > 0) {
-        mw <- pmax(w[dd], w[dd + 1])
-        w[dd] <- mw
-        w[dd + 1] <- mw
-        el <- el[-dd, , drop = FALSE]
-        w <- w[-dd]
-      }
-    }
-    el <- cbind(el, w)
-  } else if (mode == "upper") {
-    ## UPPER
-    if (diag) {
-      adjmatrix <- Matrix::triu(adjmatrix)
-    } else {
-      adjmatrix <- Matrix::triu(adjmatrix, 1)
-    }
-    el <- mysummary(adjmatrix)
-    rm(adjmatrix)
-    if (!diag) {
-      el <- el[el[, 1] != el[, 2], ]
-    }
+    adjmatrix <- Matrix::tril(adjmatrix)
   } else if (mode == "lower") {
-    ## LOWER
-    if (diag) {
-      adjmatrix <- Matrix::tril(adjmatrix)
-    } else {
-      if (vc == 1) {
-        # Work around Matrix glitch
-        adjmatrix <- as(matrix(0), "dgCMatrix")
-      } else {
-        adjmatrix <- Matrix::tril(adjmatrix, -1)
-      }
-    }
-    el <- mysummary(adjmatrix)
-    rm(adjmatrix)
-    if (!diag) {
-      el <- el[el[, 1] != el[, 2], ]
-    }
-  } else if (mode == "min") {
-    ## MINIMUM
-    adjmatrix <- sign(adjmatrix) * sign(Matrix::t(adjmatrix)) * adjmatrix
-    el <- mysummary(adjmatrix)
-    rm(adjmatrix)
-    if (!diag) {
-      el <- el[el[, 1] != el[, 2], ]
-    }
-    el <- el[el[, 3] != 0, ]
-    w <- el[, 3]
-    el <- el[, 1:2]
-    el <- cbind(pmin(el[, 1], el[, 2]), pmax(el[, 1], el[, 2]))
-    o <- order(el[, 1], el[, 2])
-    el <- el[o, ]
-    w <- w[o]
-    if (nrow(el) > 1) {
-      dd <- el[2:nrow(el), 1] == el[1:(nrow(el) - 1), 1] &
-        el[2:nrow(el), 2] == el[1:(nrow(el) - 1), 2]
-      dd <- which(dd)
-      if (length(dd) > 0) {
-        mw <- pmin(w[dd], w[dd + 1])
-        w[dd] <- mw
-        w[dd + 1] <- mw
-        el <- el[-dd, ]
-        w <- w[-dd]
-      }
-    }
-    el <- cbind(el, w)
+    adjmatrix <- Matrix::tril(adjmatrix)
+  } else if (mode == "upper") {
+    adjmatrix <- Matrix::triu(adjmatrix)
   } else if (mode == "plus") {
-    ## PLUS
     adjmatrix <- adjmatrix + Matrix::t(adjmatrix)
-    if (diag) {
-      adjmatrix <- Matrix::tril(adjmatrix)
-    } else {
-      if (vc == 1) {
-        # Work around Matrix glitch
-        adjmatrix <- as(matrix(0), "dgCMatrix")
-      } else {
-        adjmatrix <- Matrix::tril(adjmatrix, -1)
-      }
-    }
-    el <- mysummary(adjmatrix)
-    rm(adjmatrix)
-    if (diag) {
-      loop <- el[, 1] == el[, 2]
-      el[loop, 3] <- el[loop, 3] / 2
-    }
-    el <- el[el[, 3] != 0, ]
+    adjmatrix <- Matrix::tril(adjmatrix)
+  } else if (mode == "max") {
+    adjmatrix <- pmax_AtA(adjmatrix, Matrix::t(adjmatrix))
+    adjmatrix <- Matrix::tril(adjmatrix)
+  } else if (mode == "min") {
+    adjmatrix <- pmin_AtA(adjmatrix, Matrix::t(adjmatrix))
+    adjmatrix <- Matrix::tril(adjmatrix)
+    adjmatrix <- Matrix::drop0(adjmatrix)
+  }
+  el <- mysummary(adjmatrix)
+  if (!diag) {
+    el <- el[el[, 1] != el[, 2]]
```

Comment:
Do we need a comma?

```suggestion
    el <- el[el[, 1] != el[, 2], ]
```

Reply:
yes is needed and was not covered by a test. Now it is